apps/condo/domains/notification/schema/SendMessageService.js

```diff
@@ -0,0 +1,123 @@
+}
+
+async function checkSendMessageMeta (type, meta) {
+    if (meta.dv !== 1) throw new Error(`${JSON_UNKNOWN_VERSION_ERROR}meta] Unknown \`dv\` attr inside JSON Object`)
+    const schema = MESSAGE_META[type]
+    if (!schema) throw new Error('unsupported type or internal error')
+    for (const attr of Object.keys(schema)) {
+        const value = meta[attr]
+        const { required } = schema[attr]
+        if (required && !value) throw new Error(`${JSON_NO_REQUIRED_ATTR_ERROR}meta] no ${attr} value`)
+    }
+    for (const attr of Object.keys(meta)) {
+        if (!ALPHANUMERIC_REGEXP.test(attr)) throw new Error(`${JSON_SUSPICIOUS_ATTR_NAME_ERROR}meta] unsupported attr name charset`)
+        if (!schema[attr]) throw new Error(`${JSON_UNKNOWN_ATTR_NAME_ERROR}meta] ${attr} is redundant or unknown`)
+    }
+}
+
+const SendMessageService = new GQLCustomSchema('SendMessageService', {
+    types: [
+        {
+            access: true,
+            type: `enum SendMessageLang { ${Object.keys(LOCALES).join(' ')} }`,
+        },
+        {
+            access: true,
+            type: 'input SendMessageToInput { user: UserWhereUniqueInput, email: String, phone: String }',
+        },
+        {
+            access: true,
+            type: `enum SendMessageType { ${MESSAGE_TYPES.join(' ')} }`,
+        },
+        {
+            access: true,
+            type: 'input SendMessageInput { dv: Int!, sender: JSON!, to: SendMessageToInput!, type: SendMessageType!, lang: SendMessageLang!, meta: JSON!, organization: OrganizationWhereUniqueInput }',
+        },
+        {
+            access: true,
+            type: 'type SendMessageOutput { status: String!, id: String! }',
+        },
+        {
+            access: true,
+            type: 'input ResendMessageInput { dv: Int!, sender: JSON!, message: MessageWhereUniqueInput }',
+        },
+        {
+            access: true,
+            type: 'type ResendMessageOutput { status: String!, id: String! }',
+        },
+    ],
+    mutations: [
+        {
+            access: access.canSendMessage,
+            schema: 'sendMessage(data: SendMessageInput!): SendMessageOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                // TODO(pahaz): think about sending emails with attachments
+                const { data } = args
+                const { dv, sender, to, type, meta, lang } = data
+                if (!to.user && !to.email && !to.phone) throw new Error('invalid send to input')
+
+                await checkSendMessageMeta(type, meta)
+
+                const messageAttrs = { dv, sender, status: 'sending', type, meta, lang }
+                if (to.email) messageAttrs.email = to.email
+                if (to.phone) messageAttrs.phone = to.phone
+                if (to.user) messageAttrs.user = { connect: to.user }
+
+                const message = await Message.create(context, messageAttrs)
+
+                await scheduleMessageDelivery(message)
```

Comment:
What about exceptions handle from  `sendMessageByTransport`, should we handle them?

Reply:
I'm not sure. At the moment it's OK if we throw the error up to GQL.
If some error accrued during the scheduleMessageDelivery call it meant the sendMessage should fail too.

Reply:
And for transports exceptions, we handle all transport exceptions.